python/src/trezorlib/cli/cardano.py

```diff
@@ -40,20 +40,27 @@ def cli():
 @cli.command()
 @click.argument("file", type=click.File("r"))
 @click.option("-f", "--file", "_ignore", is_flag=True, hidden=True, expose_value=False)
+@click.option(
+    "-s",
+    "--signing-mode",
+    required=True,
+    type=click.Choice(cardano.SIGNING_MODE_IDS.keys(), case_sensitive=False),
```

Comment:
instead:
```
ChoiceType({m.name: m for m in messages.CardanoTxSigningMode})
```
(modify `ChoiceType` if you need the `case_sensitive` flag?)
then you can remove `SIGNING_MODE_IDS` entirely

Reply:
:+1: https://github.com/trezor/trezor-firmware/commit/45b53e8adebc8240d508e07df799b5e1a1de3ee8 The `case_sensitive` flag is not really needed.